src/actions/actionHelpers.js

```diff
@@ -1,33 +1,44 @@
 import { authenticatedFetch, authenticatedPost, authenticatedPut } from 'services/authenticatedFetch'
 
+export const testableReduxBackedPromise = (localAuthenticatedFetch, localAuthenticatedPost, localAuthenticatedPut) =>
+(promiseArgs, actionSet, operation = 'GET') =>
+(dispatch) => {
+  validateActionSet(actionSet)
+
+  const promiseGenerator = getPromiseGenerator(localAuthenticatedFetch, localAuthenticatedPost, localAuthenticatedPut)(operation)
+
+  dispatch(actionSet.try())
+
+  return promiseGenerator(dispatch, ...promiseArgs)
+        .then(({json, response}) => dispatch(actionSet.succeed(json, response)),
+            error => dispatch(actionSet.fail(error)))
+}
+
+export const reduxBackedPromise = testableReduxBackedPromise(authenticatedFetch, authenticatedPost, authenticatedPut)
+
 const needOnActionSet = (prop) => `Need "${prop}" function on actionSet!`
 
-export const reduxBackedPromise = (promiseArgs, actionSet, operation = 'GET') => (dispatch) => {
-  if (!actionSet.try) { throw needOnActionSet('try') }
-  if (!actionSet.succeed) { throw needOnActionSet('succeed') }
-  if (!actionSet.fail) { throw needOnActionSet('fail') }
+export const validateActionSet = (actionSet) => {
+  let need
+  if (!actionSet.fail || !(typeof actionSet.fail === 'function')) { need = 'fail' }
+  if (!actionSet.succeed || !(typeof actionSet.succeed === 'function')) { need = 'succeed' }
+  if (!actionSet.try || !(typeof actionSet.try === 'function')) { need = 'try' }
+  if (need) { throw needOnActionSet(need) }
```

Comment:
throwing an Error instead of string is usually preferable:
  `throw new Error(needOnActionSet(need))`

Depending on the browser, it may include additional contextual information, and has better cross-browser compatibility.